### DISCUSSION TITLE
Regarding Stoping developer to commit the secrete and API and password

### DISCUSSION THREAD

**Discussion by bharatxnj** [zricethezav/gitleaks#528] [Q&A] [answered]
Hello Guys, 

Just Curious to know, Is there any automation way, i can prevent developer to stop committing Secrete, API keys, USER Name and Password directly on Git-labs. When they commit on git-labs using git-leaks software. 

Currently, how i am using Git-leaks, Cloning the Git-labs repo on my local, Using the rules of TOMAL file. I scan my repo. I am getting the result also in CSV file format.  I have to do this manually, looking for automation way. 

**_ISSUE_**:-->i am looking for automation solution in which developer not able to commit such type practices like committing Secrete, API keys, User Name and Password. 

Is there any kind of Git-leaks hooks which connect to Git-labs. 

Is this possible on GITLEAKS of automation work?

Please guide me on it.

Thanks
Bharat

**Comment by w0rmr1d3r** [comment, marked as answer]
Hello Bharat!
As said in the documentation, you can use it in an automation tool by running it, for example, as a docker or with Pre-Commit:
https://github.com/zricethezav/gitleaks#installation

Then, you an use it as well as a GitHub Action by using this repo -> https://github.com/zricethezav/gitleaks-action

Cheers!
